packages/ui/lib/components/Display/Address.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect } from 'react'
+import {
+  FiCopy as CopyIcon,
+  FiExternalLink as ExternalLinkIcon,
+} from 'react-icons/fi'
+import useEns from '../../../../../unlock-app/src/hooks/useEns'
+import { Placeholder } from '../Placeholder'
+import useClipboard from 'react-use-clipboard'
+import { ToastHelper } from '../../../../../unlock-app/src/components/helpers/toast.helper'
+import { minifyAddress } from '~/utils'
+
+export interface AddressProps {
+  address: string
+  showExternalLink?: boolean
+  showCopyIcon?: boolean
+  showENSName?: boolean
+  className?: string
+  minified?: boolean
+}
+
+// Address component
+export const Address: React.FC<AddressProps> = ({
+  address,
+  showExternalLink = false,
+  showCopyIcon = false,
+  showENSName = true,
+  className = '',
+  minified = true,
+}) => {
+  const resolvedName = useEns(address)
+  const displayName =
+    showENSName && resolvedName !== address
+      ? resolvedName
+      : minified
+        ? minifyAddress(address)
+        : address
+
+  const [isCopied, setCopy] = useClipboard(address)
+
+  useEffect(() => {
+    if (!isCopied) return
+    ToastHelper.success('Address copied')
```

Comment:
Instead of showing a toast here, let's just have a `onCopied` prop that gets called when copied. This way the app can decide what to do and what message to display if applicable.